### PR TITLE
Media : Clean folder structure

### PR DESCRIPTION
**Pull request by Taejun-Kwon**
1. Remove MediaUtils.h from framework/include/media
2. Create media folders to maintain media utils
3. clean include codes related to MediaUtils.h
4. Add MediaTypes.h to provide Media Type instead of MediaUtils.h

### CODE REVIEW

File: framework/src/media/FileInputDataSource.cpp [hunk 1]
```diff
@@ -63,14 +64,14 @@ bool FileInputDataSource::open()
 	setAudioType(utils::getAudioTypeFromPath(mDataPath));
 
 	switch (getAudioType()) {
-		case utils::AUDIO_TYPE_MP3:
-		case utils::AUDIO_TYPE_AAC:
-			setDecoder(std::make_shared<Decoder>());
+		case AUDIO_TYPE_MP3:
+		case AUDIO_TYPE_AAC:
+			setDecoder(std::make_shared<Decoder>(new Decoder()));
```
Comment (jsdosa) [inline framework/src/media/FileInputDataSource.cpp]: remove new Decoder()

**Comment by seinfra** [comment]
Target : [385e5d79e61466f70f75922d7791d7e3fe2cc737] - Code Rule Check Result:
framework/include/media/DataSource.h:78: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/DataSource.h:78: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/FileInputDataSource.h:42: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/FileInputDataSource.h:63: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/FileInputDataSource.h:63: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:69: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:75: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:75: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/FileInputDataSource.h:75: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:122: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:45: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/InputDataSource.h:59: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/InputDataSource.h:65: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/InputDataSource.h:65: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/InputDataSource.h:65: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/InputDataSource.h:72: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/include/media/InputDataSource.h:73: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/media/InputDataSource.h:73: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/media/InputDataSource.h:73: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/include/media/InputDataSource.h:80: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/media/InputDataSource.h:80: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/media/InputDataSource.h:80: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/include/media/InputDataSource.h:93: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:93: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:93: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:96: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:96: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:96: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:96: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:102: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:102: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:102: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/utils/MediaUtils.h:44: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/utils/MediaUtils.h:44: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/utils/MediaUtils.h:44: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/utils/MediaUtils.h:51: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/utils/MediaUtils.h:51: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/utils/MediaUtils.h:51: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/utils/MediaUtils.h:59: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/utils/MediaUtils.h:59: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)

**Comment by seinfra** [comment]
Target : [385e5d79e61466f70f75922d7791d7e3fe2cc737] - Code Rule Check (C++) OK.

File: framework/src/media/FileInputDataSource.cpp (hunk 1 above)
Comment (Taejun-Kwon) [reply framework/src/media/FileInputDataSource.cpp]: Thanks, maybe it overwritten.

**Comment by seinfra** [comment]
Target : [7e469703fc828318d3cd3263b37c84558d572b49] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7e469703fc828318d3cd3263b37c84558d572b49] - Code Rule Check Result:
framework/include/media/DataSource.h:78: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/DataSource.h:78: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/FileInputDataSource.h:42: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/FileInputDataSource.h:63: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/FileInputDataSource.h:63: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:69: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:75: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:75: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/FileInputDataSource.h:75: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/FileInputDataSource.h:122: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:45: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/InputDataSource.h:59: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/InputDataSource.h:65: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/InputDataSource.h:65: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/InputDataSource.h:65: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/InputDataSource.h:72: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/include/media/InputDataSource.h:73: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/media/InputDataSource.h:73: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/media/InputDataSource.h:73: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/include/media/InputDataSource.h:80: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/media/InputDataSource.h:80: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/media/InputDataSource.h:80: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/include/media/InputDataSource.h:93: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:93: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:93: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:96: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:96: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:96: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:96: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:102: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:102: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:102: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/utils/MediaUtils.h:44: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/utils/MediaUtils.h:44: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/utils/MediaUtils.h:44: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/utils/MediaUtils.h:51: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/utils/MediaUtils.h:51: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/utils/MediaUtils.h:51: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/utils/MediaUtils.h:59: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/utils/MediaUtils.h:59: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)

**Pull request merged**